File: samples.vb/AdventureWorksLT.SqlServer/AdventureWorksLT.DbInit/MockEmptySalesOrder.vb
Imports System.Threading
Imports DevZest.Data

''' <remarks><see cref="SalesOrder"/> And <see cref="SalesOrderDetail"/> are chosen for having foreing key to non-existing table(s) And
''' parent-child relationship.</remarks>
Public Class MockEmptySalesOrder
    Inherits DbMock(Of Db)

    Public Shared Function CreateAsync(db As Db, Optional progress As IProgress(Of DbInitProgress) = Nothing, Optional ct As CancellationToken = Nothing) As Task(Of Db)
        Return New MockEmptySalesOrder().MockAsync(db, progress, ct)
    End Function

    ' The order of mocking table does Not matter, the dependencies will be sorted out automatically.
    Protected Overrides Sub Initialize()
        Mock(Db.SalesOrderDetail)
        Mock(Db.SalesOrderHeader)
    End Sub
End Class

File: samples/AdventureWorksLT.MySql/AdventureWorksLT.DbInit/MockEmptySalesOrder.cs
using DevZest.Data;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace DevZest.Samples.AdventureWorksLT
{
    /// <remarks><see cref="SalesOrder"/> and <see cref="SalesOrderDetail"/> are chosen for having foreing key to non-existing table(s) and
    /// parent-child relationship.</remarks>
    public sealed class MockEmptySalesOrder : DbMock<Db>
    {
        public static Task<Db> CreateAsync(Db db, IProgress<DbInitProgress> progress = null, CancellationToken ct = default(CancellationToken))
        {
            return new MockEmptySalesOrder().MockAsync(db, progress, ct);
        }

        protected override void Initialize()
        {
            // The order of mocking table does not matter, the dependencies will be sorted out automatically.
            Mock(Db.SalesOrderDetail);
            Mock(Db.SalesOrderHeader);
        }
    }
}
